lib/Service/MailManager.php

```diff
@@ -386,14 +387,15 @@ public function flagMessage(Account $account, string $mailbox, int $uid, string
 		}
 
 		// Only send system flags to the IMAP server as other flags might not be supported
-		$imapFlags = self::ALLOWED_FLAGS[$flag] ?? [];
+		$imapFlags = $this->filterFlags($account, $flag, $mailbox);
 		try {
 			foreach ($imapFlags as $imapFlag) {
-				if ($value) {
-					$this->imapMessageMapper->addFlag($client, $mb, $uid, $imapFlag);
-				} else {
-					$this->imapMessageMapper->removeFlag($client, $mb, $uid, $imapFlag);
+				if (empty($imapFlag) === true) {
+					continue;
 				}
+				($value === true)
+					? $this->imapMessageMapper->addFlag($client, $mb, $uid, $imapFlag)
+					: $this->imapMessageMapper->removeFlag($client, $mb, $uid, $imapFlag);
```

Comment:
I'm not a big fan of this operator. The readability of this block could be improved by using a regular if.

Reply:
I agree. Ternary is fine for an expression (anything that has a result) but it's strange for methods with no returned value

Reply:
I'll rework this